pre_commit/languages/conda.py

```diff
@@ -26,6 +25,9 @@ def install_environment(prefix, version, additional_dependencies):
 
 def run_hook(hook, file_args, color):
     directory = helpers.environment_dir(ENVIRONMENT_DIR, hook.language_version)
-    env_dir = hook.prefix.path(directory)
-    cmd = ('conda', 'run', '-p', env_dir) + hook.cmd
+    cmd = (hook.prefix.path(directory, 'bin', hook.cmd[0]),) + hook.cmd[1:]
```

Comment:
using the env patch and prefixing `PATH` is a more reliable way to accomplish this -- though if `conda run` works then maybe we just use that?

Reply:
`conda run` produces some annoying output in the failure case and depending on the environment introduces a significant runtime overhead. I would make that switchable on a per-hook basis. What would be the best approach for that? Have two languages  `conda` and `conda_run`?

With env patch and prefixing, you mean `with envcontext(get_env_patch(envdir)):`?

Reply:
maybe for now we just make `conda` and don't enable `conda run` until someone asks for it (?) -- and yes to `envcontext` :+1: